### ISSUE TITLE
how to rebuild?

### ISSUE THREAD

**Issue by zjorgensenbits** [ChrisChross/streamlit-agraph#16]
What's the proper procedure for rebuilding after modifying one of the tsx files?

**Comment by zjorgensenbits** [comment]
nevermind, I figured it out:  run the following from the frontend directory
npm install
npm run build

**Comment by ChrisChross** [comment]
Would be interesting what changes you made.
Maybe we can merge them?

**Comment by zjorgensenbits** [comment]
I modified AgraphComponent.tsx to pass the ID of the clicked node back to python through the return value. I can try to clean it up and submit a pull request, but all I really did was add `Streamlit.setComponentValue(nodeId);` to the click handlers, but it seems to work well.

**Comment by ChrisChross** [comment]
Great, would be great if you could merge it!
Also thanks for going along with the other open issue 👍 .

**Comment by AlliedToasters** [comment]
Hi @zjorgensenbits 
You are my hero.

**Comment by jnferfer** [comment]
Hi @zjorgensenbits 
I've also modified one of the tsx files in a forked version of the agraph repo (https://github.com/jnferfer/streamlit-agraph)
Now I want to use this version in app deployment, but I don't know how to do it. In `requirements.txt` I've added the following line:
`git+https://github.com/jnferfer/streamlit-agraph.git@master`
But I get an error when deploying the app to Streamlit cloud. How did you manage to deploy? Thanks!
